soc/arm/st_stm32/stm32h7/Kconfig.defconfig.stm32h7a3xx

```diff
@@ -6,7 +6,8 @@
 if SOC_STM32H7A3XX
 
 config SOC
-	default "stm32h7a3xx"
+	default "stm32h7a3xxQ" if POWER_SUPPLY_SMPS
+	default "stm32h7a3xx" if !POWER_SUPPLY_SMPS
```

Comment:
This is conceptually wrong as this is selecting the HW (well CMSIS file description) based on SW configuration.
This link between `POWER_SUPPLY_SMPS` and HW reference should be reworked

Reply:
The alternate idea will be defining an `SMPS` node on the device tree, and the base SoC is stm32h7a3xx SoC, but then it won't  point to the exact CMIS file

What will be a better way of accomplishing it?

Reply:
Ideal way would be device tree configuration, but for now we're missing the proper regulator minimum description to support this (at least for current PR).
Problem comes from `POWER_SUPPLY_SMPS`:
- it is user-selectable
- description: `Enable SMPS Power Supply.`

So either you need to add an additional Kconfig symbol, non user-selectable (promptless), like `HAS_POWER_SUPPLY_SMPS`, which would select the corect CSMSIS file and implies `POWER_SUPPLY_SMPS` either `POWER_SUPPLY_SMPS` definition should be reviewed.

Reply:
Sorry don't understand the following:

- `HAS_POWER_SUPPLY_SMPS` is set by who and where?

The idea is `POWER_SUPPLY_SMPS` acts as user/board-level decision input to enable/disable the SMPS feature, so what is wrong with the current description?

Reply:
The whole system ends up selecting a SoC description (`stm32h7a3xx` or `stm32h7a3xxQ`).
This is wrong, because SoC is not configurable but the base of the target configuration:
The SW configuration should be a consequence of the SoC selection and not the other way round.

Reply:
> 

Please take a look at my last temp commit. Does that meet all the requirements?

Reply:
@erwango, mind taking a look?

Reply:
Hi @erwango, any update regarding this PR?